portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -1978,6 +1981,32 @@ public void includeLogin(PortalRenderContext rcontext, HttpServletRequest req,
 				rcontext.put("loginUserFirstName", loginUserFirstName);
 				rcontext.put("loginUserDispId", loginUserDispId);
 				rcontext.put("loginUserId", loginUserId);
+
+				boolean displayRole = ServerConfigurationService.getBoolean(PORTAL_DISPLAY_CURRENT_ROLE, true);
+				rcontext.put("actualRole", "");
+				if(displayRole){
+					String[] parts = getParts(req);
+					String roleId = "";
+					String siteId = (parts.length > 2)? parts[2]: "";
+					if (siteId != "" && siteId.indexOf("~") != 0){
+						try{
+							Site site = SiteService.getSite(siteId);
+							roleId = site.getMember(loginUserId).getRole().getId();
+						}catch (Exception e){
+							log.warn(e.getMessage());
+						}
+					}
+					String roleMessage = "";
+					if(roleId.equals("Instructor")) {
```

Comment:
Just return the roleId....

rcontext.put("currentRole", rloader.getFormattedMessage("sit_role", roleId));